bin/run-utils.js

```diff
@@ -0,0 +1,192 @@
+ * @return {Boolean}
+ */
+function isErrorString(line) {
+  return /^\*{25}/.test(line) || /^\s*Message: [\D]*Error/.test(line);
+}
+
+var GARBAGE = [
+  /\[JavaScript Warning: "TypeError: "[\w\d]+" is read-only"\]/,
+  /JavaScript strict warning: /,
+  /\#\#\#\!\!\! \[Child\]\[DispatchAsyncMessage\]/,
+  /JavaScript strict warning: resource:\/\/\/modules\/sessionstore\/SessionStore\.jsm/,
+  /JavaScript strict warning: resource:\/\/gre\/components\/nsSearchService\.js/
+];
+
+/**
+ * Checks of the chunk of process output can be discarded. The patterns that are
+ * tested against can be found in the `GARBAGE` array above.
+ *
+ * @param  {String} data
+ * @return {Boolean}
+ */
+function isGarbage(data) {
+  if (!data || !data.replace(/[\s\t\r\n]+/, "")) {
+    return true;
+  }
+  for (var i = 0, l = GARBAGE.length; i < l; ++i) {
+    if (GARBAGE[i].test(data)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * Output a chunk of process output to the console (Terminal).
+ *
+ * @param {String} data
+ * @param {String} type Type of output; can be 'log' (default), 'warn' or 'error'
+ */
+function writeLog(data, type) {
+  if (isGarbage(data)) {
+    return;
+  }
+  process[(type === "error") ? "stderr" : "stdout"].write(data);
+}
+
+/**
+ * Takes options, and runs Firefox. Inspiration of this code can be found at
+ * https://github.com/mozilla-jetpack/jpm.
+ *
+ * @param {Object} options
+ *   - `binary` path to Firefox binary to use
+ *   - `profile` path to profile or profile name to use
+ *   - `binaryArgs` binary arguments to pass into Firefox, split up by spaces
+ *   - `verbose` whether or not Firefox should print all of stdout
+ * @return {Promise}
+ */
+function runFirefox(options) {
+  return When.promise(function(resolve, reject) {
+    options = options || {};
+
+    FxRunner({
+      "binary": options.binary,
+      "foreground": true,
+      "profile": options.profile,
+      env: Extend({}, process.env, FX_ENV),
+      verbose: true,
+      "binary-args": options.binaryArgs
+    }).then(function(results) {
+      var firefox = results.process;
+
+      console.log("Executing Firefox binary: " + results.binary);
+      console.log("Executing Firefox with args: " + results.args);
+
+      firefox.on("error", function(err) {
+        if (/No such file/.test(err) || err.code === "ENOENT") {
+          console.error("No Firefox binary found at " + results.binary);
+          if (!options.binary) {
+            console.error("Specify a Firefox binary to use with the `-b` flag.");
+          }
+        } else {
+          console.error(err);
+        }
+        reject(err);
+      });
+
+      firefox.on("close", resolve);
+
+      firefox.stderr.setEncoding("utf8");
+      firefox.stderr.on("data", function(data) {
+        // Only print out annoying warnings if verbose is on
+        if (/^\s*System JS : WARNING/.test(data) && options.verbose) {
+          writeLog(data, "warn");
+        } else if (options.verbose) {
```

Comment:
It looks like these `options.verbose` checks always end up being false but we /do/ have FxRunner verbose: true always.